### PR TITLE
[docs] Add possible remedy for Hermes debugger problems when the project is not served on LAN connection

### PR DESCRIPTION
**Pull request by Nantris**
To clarify: the remedy works, just can't guarantee it works for everyone.

# Why

Hermes troubleshooting steps helped to uncover the problem, but not solve it.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin). (N/A)

### CODE REVIEW

File: docs/pages/guides/using-hermes.mdx [hunk 1]
```diff
@@ -77,7 +77,7 @@ Alternatively, you can use the JavaScript inspector from the following tools:
 - Internally, the app will establish a WebSocket connection, make sure your app is connected to the development server.
 
   - Try to reload the app by pressing <kbd>r</kbd> in the Expo CLI Terminal UI.
-  - Test debugging availability by running the command: `curl http://127.0.0.1:19000/json/list` (adjust the `127.0.0.1:19000` to match your dev server URL). The HTTP response should not be an empty array, but rather something like this:
+  - Test debugging availability by running the command: `curl http://127.0.0.1:19000/json/list` (adjust the `127.0.0.1:19000` to match your dev server URL). The HTTP response should not be an empty array, but rather something like below. If it is empty, try adding one of `--localhost` or `--tunnel` to the `start` command.
```
Comment (amandeepmittal) [inline docs/pages/guides/using-hermes.mdx:80]:
```suggestion
  - Test debugging availability by running the command: `curl http://127.0.0.1:19000/json/list` (adjust the `127.0.0.1:19000` to match your dev server URL). The HTTP response should be an array, as shown below. If it is an empty response, add either the `--localhost` or `--tunnel` flag to the `npx expo start` command.
```

A minor suggestion to update the verbiage here.

**Review by amandeepmittal — COMMENTED** [review]
Thanks for adding the remedy. Left a small suggestion to have consistent verbiage.

**Comment by Slapbox** [comment]
@amandeepmittal thanks for the edit. I've found it may also be necessary to `--clear-cache` once the issue has occurred on a non-working method.

ie `--lan` doesn't work, so you try `tunnel` - but `--tunnel` will only work if you use `--tunnel --clear-cache` (but `--clear-cache` alone seems to have no effect.) I'm not sure how to update the wording to reflect that new finding. Any suggestion?

**Comment by amandeepmittal** [comment]
@Slapbox, I don't think `--clear-cache` is a valid flag for `npx expo start`. Are you using the local versioned CLI that comes with Expo package in your project?

**Comment by Slapbox** [comment]
> Are you using the local versioned CLI that comes with Expo package in your project?

Yes, via `yarn expo start`

**Comment by amandeepmittal** [comment]
The flag is `--clear`. You can also check that by running `yarn expo start --help` inside the Expo project.

Here is an output of that command:
CleanShot 2023-01-19 at 22 22 27@2x

I'm not sure why you are running `--clear-cache` and why is it working in your case. I also don't think `-c` or `--clear` is necessary with `--tunnel`. The purpose of the flag is when there are  changes`babel.config.js` file or when some things are done with symlinks, in those cases, the flag is required.

**Comment by Slapbox** [comment]
Not sure what to say. I can only report what has worked here. Perhaps passing an invalid flag somehow influences whether the debugger can connect or not. Or maybe I was using `--clear` and I'm conflating it with `--clear-cache` in EAS.

I'll loop back next time I run into this issue and report what worked.

**Comment by amandeepmittal** [comment]
No worries at all! Feel free to open an issue if you find anything weird 😄 

I'm going to merge this with the current changes. Thanks for your contribution!!!

**Review by amandeepmittal — APPROVED** [review]

**Comment by Slapbox** [comment]
@amandeepmittal, just an update: these days I'm noticing `yarn expo start --lan` almost always works without any other flags needed. I never need to clear the cache anymore for it to work but I always need `--lan` or I get the message about the debugger only working with Hermes apps.

**Pull request merged**